astropy/stats/funcs.py

```diff
@@ -790,3 +792,322 @@ def mad_std(data):
+    -------
+    breaks : array of floats length N
+        The endpoints of a set of intervals covering [0,1]; breaks[0]=0 and
+        breaks[-1] = 1
+    weights : array of floats of length N-1
+        The ith element is the sum of number of times the interval 
+        breaks[i],breaks[i+1] is included in each interval times the weight
+        associated with that interval.
+
+    """
+    r = []
+    breaks = set()
+    tot = 0
+    for (a,b,wt) in intervals:
+        tot += (np.ceil(b)-np.floor(a))*wt
+        fa = a%1
+        breaks.add(fa)
+        r.append((0,fa,-wt))
+        fb = b%1
+        breaks.add(fb)
+        r.append((fb,1,-wt))
+
+    breaks.add(0.)
+    breaks.add(1.)
+    breaks = list(breaks)
+    breaks.sort()
+    breaks_map = dict([(f,i) for (i,f) in enumerate(breaks)])
+    totals = np.zeros(len(breaks)-1)
+    totals += tot
+    for (a,b,wt) in r:
+        totals[breaks_map[a]:breaks_map[b]]+=wt
+    return np.array(breaks), totals
+
+def cdf_from_intervals(breaks, totals):
+    """Construct a callable piecewise-linear CDF from a pair of arrays.
+    
+    Take a pair of arrays in the format returned by fold_intervals and
+    make a callable cumulative distribution function on the interval
+    (0,1).
+
+    Parameters
+    ----------
+    breaks : array of floats of length N
+        The boundaries of successive intervals.
+    weights : array of floats of length N-1
+        The weight for each interval.
+
+    Returns
+    -------
+    f : callable
+        A cumulative distribution function corresponding to the 
+        piecewise-constant probability distribution given by breaks, weights
+
+    """
+    if breaks[0]!=0 or breaks[-1]!=1:
+        raise ValueError("Intervals must be restricted to [0,1]")
+    if np.any(np.diff(breaks)<=0):
+        raise ValueError("Breaks must be strictly increasing")
+    if np.any(totals<0):
+        raise ValueError("Total weights in each subinterval must be nonnegative")
+    if np.all(totals==0):
+        raise ValueError("At least one interval must have positive exposure")
+    b = breaks.copy()
+    c = np.concatenate(((0,), np.cumsum(totals*np.diff(b))))
+    c /= c[-1]
+    def cdf(x):
+        ix = np.searchsorted(b[:-1],x)
+        l, r = b[ix-1], b[ix] 
+        return ((r-x)*c[ix-1]+(x-l)*c[ix])/(r-l)
+    return cdf
+
+def interval_overlap_length(i1,i2):
+    """Compute the length of overlap of two intervals.
+    
+    Parameters
+    ----------
+    i1, i2 : pairs of two floats
+        The two intervals.
+
+    Returns
+    -------
+    l : float
+        The length of the overlap between the two intervals.
+    
+    """
+    (a,b) = i1
+    (c,d) = i2
+    if a<c:
+        if b<c:
+            return 0.
+        elif b<d:
+            return b-c
+        else:
+            return d-c
+    elif a<d:
+        if b<d:
+            return b-a
+        else:
+            return d-a
+    else:
+        return 0
+
+def histogram_intervals(n, breaks, totals):
```

Comment:
No. The idea is to record for each histogram bin, how much it is covered by the intervals. This is not what np.histogram does.